### PR TITLE
Adding CircleCI

### PR DESCRIPTION
**Pull request by jodiedoubleday**
#### What does this PR do? (please provide any background)
We may be moving from TravisCI to CircleCI - This PR add in the circle.yml file and has the same configuration as already exists in Travis.

#### What tests does this PR have?
No tests, however i pushed a working branch, a broken branch and a fixed branch to check everything worked as expected.

#### How can this be tested?
- Pull down the UI-Toolkit
- Create a branch
- Push the branch and check everything is okay in CircleCI
- Break a couple of tests an push the branch
- Check CircleCI fails (should show as failed in build chat in slack)
- Fixed the tests again and push
- Check CircleCI passes (should also see fix in build chat in slack)

#### Screenshots / Screencast
screen shot 2015-11-06 at 14 23 42

#### What gif best describes how you feel about this work?
circle

---
#### Developer Definition of Done/Quality Checklist (for PR author to complete BEFORE code review):
- [x] I have checked the [contributing document](../blob/master/CONTRIBUTING.md) and I'm happy for this to be reviewed.

#### Software Engineer or Developer review:
- [x] I’ve witnessed the work behaving as expected (this could be on the authors machine or screencast).
- [x] I’ve checked for coding anti-patterns.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked all the tests are passing.

#### Software Engineer or project guru review:
- [x] I’ve witnessed the work behaving as expected (this could be on the authors machine or screencast).
- [x] I’ve checked for coding anti-patterns.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked all the tests are passing.

### CODE REVIEW

**Comment by DuncanFenning** [comment]
Taking dev

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+machine:
+  node:
+    version: 0.10.40
+dependencies:
+  cache_directories:
+    - node_modules
+test:
+  override:
+    - npm run ci > $CIRCLE_TEST_REPORTS/results.xml
```
Comment (hx-markterry) [inline circle.yml]: can we change this to just `npm run ci > $CIRCLE_TEST_REPORTS/results.xml`

File: package.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
   },
   "scripts": {
     "build": "scripts/build.sh",
+    "ci": "mocha --recursive test/**/*-test.js* src/**/*-test.js* -R xunit",
```
Comment (hx-markterry) [inline package.json]: This should be all of the ci commands not just tests

File: circle.yml (hunk 1 above)
Comment (jodiedoubleday) [inline circle.yml]: I had `npm run ci` before but we wanted to push results out to xunit

File: circle.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+machine:
+  node:
+    version: 0.10.40
+dependencies:
+  cache_directories:
+    - node_modules
+test:
+  override:
+    - npm run ci
+    - npm run lint
```
Comment (hx-markterry) [inline circle.yml]: from the package.json changes below we don't need lines 9 and 10 now

File: package.json [hunk 4]
```diff
@@ -10,6 +10,7 @@
   },
   "scripts": {
     "build": "scripts/build.sh",
+    "ci": "npm run lint && npm test && npm run coverage",
```
Comment (notABluesSinger) [inline package.json]: I think the output to xunit for circle's built in reporting was a good addition and I'd like to see it back. @hx-markterry, what are your thoughts?

File: circle.yml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+machine:
+  node:
+    version: 0.10.40
+dependencies:
+  cache_directories:
+    - node_modules
+test:
+  override:
+    - npm run ci
```
Comment (notABluesSinger) [inline circle.yml]: As you are running istanbul for coverage I'd suggest adding a `post` step here and copying the coverage reports to the `CIRCLE_ARTIFACTS` folder

File: package.json (hunk 4 above)
Comment (hx-markterry) [inline package.json]: yes i agree, i'm not against xunit but was against the previous implementation of it.

**Comment by notABluesSinger** [comment]
Taking 2nd Slot

**Comment by notABluesSinger** [comment]
:+1: :shipit:

**Pull request merged**